### PR TITLE
Refactor security libs

### PR DESCRIPTION
**Pull request by davidrupprecht**
This pull request refactors the security libs: 
- It makes the snow3g independent from a global state. This mitigates a problem if the epc uses multiple user with eia1 at the same time. 
- Further I deleted the aes implementation of liblte and make all aes calls from mbedtls.

Best regards,
David

### CODE REVIEW

File: lib/src/common/snow_3g.cc [hunk 1]
```diff
@@ -1,515 +0,0 @@
-/*------------------------------------------------------------------------
- * snow_3g.c
- *
- * Adapted from ETSI/SAGE specifications:
```
Comment (andrepuschmann) [inline lib/src/common/snow_3g.cc:4]: Should we perhaps add the ETSI reference to the new file?

File: lib/include/srslte/common/s3g.h [hunk 2]
```diff
@@ -9,17 +38,11 @@
  * and Integrity Algorithms UEA2 & UIA2.
  * Document 2: SNOW 3G Specification"
  *---------------------------------------------------------*/
```
Comment (davidrupprecht) [inline lib/include/srslte/common/s3g.h:40]: @andrepuschmann I guess that is formatting wise not optional ;)

File: lib/src/common/snow_3g.cc (hunk 1 above)
Comment (davidrupprecht) [reply lib/src/common/snow_3g.cc:4]: I included it but below the includes. Shall it be directly under the license header?

**Comment by andrepuschmann** [comment]
> I included it but below the includes. Shall it be directly under the license header?

Ah, yes, missed that. I think it would be good to move it underneath our license for source and header yes. Don't worry about the `******` line, that can go away, format it as a normal comment that clang understands. What's important is the content. Thanks

**Comment by davidrupprecht** [comment]
@andrepuschmann done. Do you care much about the scope of variables? Shall I refactor this as well.

**Comment by andrepuschmann** [comment]
Yeah, why not. What did you have in mind? All under `srslte` scope?

**Comment by davidrupprecht** [comment]
Haha nope ;) But function scope makes sense for me. The suggestion for lib/src/common/s3g.cc in  https://app.codacy.com/gh/srsLTE/srsLTE/pullRequest?prid=5542730 will break the code.

**Comment by davidrupprecht** [comment]
The question is why pssch_pscch_file_test_ideal_tm2_p100 fails in the travis build. I did not touched any code.

**Comment by andrepuschmann** [comment]
> Haha nope ;) But function scope makes sense for me. The suggestion for lib/src/common/s3g.cc in https://app.codacy.com/gh/srsLTE/srsLTE/pullRequest?prid=5542730 will break the code.

Ah, didn't see that. But now I know what you mean. Yeah, sure. Since we have those fancy code analysis tools why not use it ;-)

> The question is why pssch_pscch_file_test_ideal_tm2_p100 fails in the travis build. I did not touched any code.

Weird. We had issues with flaky tests before but Travis used to be fine. Let's see if the next push will fix it ;-)

**Comment by andrepuschmann** [comment]
Now the clang-formating isn't right, so Travis complains. But the actual tests are fine ;-)

**Comment by andrepuschmann** [comment]
I've fixed the clang-format manually and added it to our internal CI/CD for checking.

**Comment by davidrupprecht** [comment]
Thanks a lot :)

**Pull request closed without merging**